docs/content/docs/deployment/security/ssl.md

```diff
@@ -0,0 +1,243 @@
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Encryption and Authentication using SSL
+
+Flink supports mutual authentication (when two parties authenticate each other at the same time) and 
+encryption of network communication with SSL for internal and external communication. 
+
+**By default, SSL/TLS authentication and encryption is not enabled** (to have defaults work out-of-the-box).
+
+This guide will explain internal vs external connectivity, and provide instructions on how to enable 
+SSL/TLS authentication and encryption for network communication with and between Flink processes. We 
+will go through steps such as generating certificates, setting up TrustStores and KeyStores, and 
+configuring cipher suites.
+
+For how-tos and tips for different deployment environments (i.e. standalone clusters, Kubernetes, YARN),
+check out the section on [Incorporating Security Features in a Running Cluster](#).
+
+## Internal and External Communication 
+
+There are two types of network connections to authenticate and encrypt: internal and external.
+
+{{< img src="/fig/ssl_internal_external.svg" alt="Internal and External Connectivity" width=75% >}}
+
+For more flexibility, security for internal and external connectivity can be enabled and configured
+separately.
+
+### Internal Connectivity
+
+Flink internal communication refers to all connections made between Flink processes. These include:
+
+- Control messages: RPC between JobManager / TaskManager / Dispatcher / ResourceManager
+- Transfers on the data plane: connections between TaskManagers to exchange data during shuffles, 
+  broadcasts, redistribution, etc
+- Blob service communication: distribution of libraries and other artifacts
+
+All internal connections are SSL authenticated and encrypted. The connections use **mutual authentication**,
+meaning both server and client side of each connection need to present the certificate to each other. 
+The certificate acts as a shared secret and can be embedded into container images or attached to your 
+deployment setup. These connections run Flink custom protocols. Users never connect directly to internal 
+connectivity endpoints.
+
+### External Connectivity
+
+Flink external communication refers to all connections made from the outside to Flink processes. 
+This includes: 
+- communication with the Dispatcher to submit Flink jobs (session clusters)
+- communication of the Flink CLI with the JobManager to inspect and modify a running Flink job/application
+
+Most of these connections are exposed via REST/HTTP endpoints (and used by the web UI). Some external 
```

Comment:
I am not certain. I saw this off some Flink training slides.  So ALL of these external connections do use these endpoints?

Reply:
AFAIK @knaufk is correct, and it would be more accurate to say "These connections are exposed via REST/HTTP endpoints (and are used by Flink's web UI). Some application-specific external services such as sources and sinks may use some other network protocol."